### ISSUE TITLE
Inconsistent results for title_exists()

### ISSUE THREAD

**Issue by sijis** [richardasaurus/imdb-pie#18]
Hi,

In testing out some functionality (imdbpie (4.0.2)), I noticed that the title_exists() method is inconsistent. I would expect that the method would either return True or False, but in some instances I receive a HTTPError.

The odd thing is that if I have enough bad characters, I get the False response, as expected. 

``` python
In [1]: import imdbpie

In [2]: imdb = imdbpie.Imdb()
..
In [53]: imdb.title_exists('tt123456')
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
<ipython-input-53-7664b9a92afc> in <module>()
----> 1 imdb.title_exists('tt123456')

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/imdbpie/imdbpie.py in title_exists(self, imdb_id)
    100             return False
    101         else:
--> 102             response.raise_for_status()
    103 
    104     def search_for_person(self, name):

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/requests/models.py in raise_for_status(self)
    849 
    850         if http_error_msg:
--> 851             raise HTTPError(http_error_msg, response=self)
    852 
    853     def close(self):

HTTPError: 405 Client Error: Method Not Allowed

In [54]: imdb.title_exists('tt1234567')
Out[54]: True

In [55]: imdb.title_exists('tt1234567890')
Out[55]: False

In [56]: 
...
```

Also, it also appears that it movie_id must begin with 'tt' otherwise it always returns the HTTPError

``` python

In [56]: imdb.title_exists('123456')
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
<ipython-input-56-ff2c3862f9a6> in <module>()
----> 1 imdb.title_exists('123456')

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/imdbpie/imdbpie.py in title_exists(self, imdb_id)
    100             return False
    101         else:
--> 102             response.raise_for_status()
    103 
    104     def search_for_person(self, name):

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/requests/models.py in raise_for_status(self)
    849 
    850         if http_error_msg:
--> 851             raise HTTPError(http_error_msg, response=self)
    852 
    853     def close(self):

HTTPError: 405 Client Error: Method Not Allowed

In [57]: imdb.title_exists('1234567')
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
<ipython-input-57-416ef5e2c95d> in <module>()
----> 1 imdb.title_exists('1234567')

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/imdbpie/imdbpie.py in title_exists(self, imdb_id)
    100             return False
    101         else:
--> 102             response.raise_for_status()
    103 
    104     def search_for_person(self, name):

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/requests/models.py in raise_for_status(self)
    849 
    850         if http_error_msg:
--> 851             raise HTTPError(http_error_msg, response=self)
    852 
    853     def close(self):

HTTPError: 405 Client Error: Method Not Allowed

In [58]: imdb.title_exists('1234567890')
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
<ipython-input-58-bdb8214432ec> in <module>()
----> 1 imdb.title_exists('1234567890')

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/imdbpie/imdbpie.py in title_exists(self, imdb_id)
    100             return False
    101         else:
--> 102             response.raise_for_status()
    103 
    104     def search_for_person(self, name):

/home/saviles/virtualenvs/errbot-py3.4.1/lib/python3.4/site-packages/requests/models.py in raise_for_status(self)
    849 
    850         if http_error_msg:
--> 851             raise HTTPError(http_error_msg, response=self)
    852 
    853     def close(self):

HTTPError: 405 Client Error: Method Not Allowed

```

**Comment by richardasaurus** [comment]
That's because you're not passing in a valid imdb_id.

a valid id is 'tt' followed by 7 digits.

e.g. 
```
tt0000000
```
The service turns away any requests which don't match that pattern, rightly so. Input validation could be added to that function shortly to raise a `ValueError` instead of going as far as the API.

**Comment by sijis** [comment]
Returning a ```ValueError``` would be useful and clearer along with the validation.

I personally do think though the method should only return True or False (unless that's not the purpose of the method) but i do understand your point of view on it.

**Comment by richardasaurus** [comment]
this is fixed in version `4.4.0`
